Sources/KeyboardShortcuts/KeyboardShortcuts.swift

```diff
@@ -407,6 +407,30 @@ extension KeyboardShortcuts {
 	- Note: This method is not affected by `.removeAllHandlers()`.
 	*/
 	@available(macOS 10.15, *)
+	public static func events(for name: Name) -> AsyncStream<KeyboardShortcuts.EventType> {
```

Comment:
You also need to add the overload with the `type` parameter, which is what most users will want to use, as you usually only need one of the events. And this makes the common case slightly more convenient.

Reply:
The overload could simply do something like: `events(for: name).filter { $0 == type }`

Reply:
I did this with the function `public static func events(_ type: EventType, for name: Name) -> AsyncFilterSequence<AsyncStream<EventType>>`, but I was unable to find a way to keep the return type to `AsyncStream<EventType>`.

`some AsyncSequence` exits but it erases the return value and needs to be called with `try`.

I looked at the evolution proposals and Swift forums but was unable to find a satisfactory alternative.

Reply:
Yeah, we need `some AsyncSequence<Element>` (hopefully coming in Swift 5.7) and rethrowing protocols.

Reply:
Alright, I'll leave it like this for now then